File: sem4/MPP/proiectC#/AppClientServer/Model/TaskDTO.cs
using System;

namespace model;
[Serializable]
public class TaskDTO:Entity<int>
{
    public string description { get; set; }
    public int ageCatStart { get; set; }
    public int ageCatEnd { get; set; }
    public int nrChildren { get; set; }

    public TaskDTO(string description, int ageCatStart, int ageCatEnd, int nrKids)
    {
        this.description = description;
        this.ageCatStart = ageCatStart;
        this.ageCatEnd = ageCatEnd;
        this.nrChildren = nrKids;
    }
    public override bool Equals(object? obj) 
    {
        if (obj is Task) 
        {       
            return (obj as Task).ID == ID; 
        } 
        return base.Equals(obj); 
    }
        
    public override string ToString() 
    {
        return ID + " " + description + " " + ageCatStart + " " + ageCatEnd + " " + nrChildren;
    }
        
    public static bool operator ==(TaskDTO t1, TaskDTO t2) 
    { 
        return t1.Equals(t2); 
    }
        
    public static bool operator !=(TaskDTO t1, TaskDTO t2) 
    { 
        return !t1.Equals(t2); 
    } 
}

File: sem4/MPP/proiect/appClientServer/CompetitionModel/src/main/java/org/example/model/TaskDTO.java
package org.example.model;

import java.io.Serializable;
import java.util.Objects;

public class TaskDTO extends Entity<Integer> implements Serializable {
    private String description;
    private int ageCatStart;
    private int ageCatEnd;
    private int noChildren;

    public TaskDTO(String description, int ageCatStart, int ageCatEnd, int noChildren) {
        this.description = description;
        this.ageCatStart = ageCatStart;
        this.ageCatEnd = ageCatEnd;
        this.noChildren = noChildren;
    }

    public TaskDTO() {
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public int getAgeCatStart() {
        return ageCatStart;
    }

    public void setAgeCatStart(int ageCatStart) {
        this.ageCatStart = ageCatStart;
    }

    public int getAgeCatEnd() {
        return ageCatEnd;
    }

    public void setAgeCatEnd(int ageCatEnd) {
        this.ageCatEnd = ageCatEnd;
    }

    public int getNoChildren() {
        return noChildren;
    }

    public void setNoChildren(int noChildren) {
        this.noChildren = noChildren;
    }

    @Override
    public String toString() {
        return "TaskDTO{" +
                "description='" + description + '\'' +
                ", ageCatStart=" + ageCatStart +
                ", ageCatEnd=" + ageCatEnd +
                ", noChildren=" + noChildren +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        TaskDTO taskDTO = (TaskDTO) o;
        return ageCatStart == taskDTO.ageCatStart && ageCatEnd == taskDTO.ageCatEnd && noChildren == taskDTO.noChildren && Objects.equals(description, taskDTO.description);
    }

    @Override
    public int hashCode() {
        return Objects.hash(description, ageCatStart, ageCatEnd, noChildren);
    }
}
